libmcount/mcount.c

```diff
@@ -860,6 +860,18 @@ static void mcount_sync_depth(struct mcount_thread_data *mtdp)
 		mtdp->filter.max_depth = mcount_depth;
 }
 
+/**
+ * mcount_sync_time - apply global time value
+ *
+ * The global time value can be changed at runtime by the agent. We check it and
+ * apply it on function entry.
+ */
+static void mcount_sync_time(struct mcount_thread_data *mtdp)
+{
+	if (!mtdp->filter.time_trigger)
+		mtdp->filter.time = mcount_threshold;
```

Comment:
Can we do this without adding new fields?  I think we can update the mtdp->filter.time only for the triggers.  And use the global value if it's not set.

Reply:
Yes, fixed that.